.gitlab-ci.yml

```diff
@@ -90,9 +90,13 @@ build:make:
 
 release-staging-ref:
   <<: *release-staging
-  when: manual
-  except:
-    - tags
+  rules:
+    - if: $CI_COMMIT_TAG
+      when: never
+    - if: $CI_COMMIT_BRANCH =~ /^.*-staging$/
+      when: on_success
+    - when: manual
+      allow_failure: true
```

Comment:
What is the rationale to allow failure here?

What are the implications?

Reply:
I set it up following [this thread](https://gitlab.com/gitlab-org/gitlab/-/issues/39534).
In short, a `rules`-defined manual job logically has to allow failure to not be considered as blocking if not run. It was not intuitive to me either, and made it work after reading the thread.
If the `allow_failure: true` line is removed, the pipeline will not skip the job by default but consider it as blocking if not run (even though it's manual). The pipeline will also put the following jobs in the `waiting` state, and will not approve the github PR CI.

Reply:
Sample of a successful job without the `allow_failure` line (it's still 'blocked' as we speak)
<img width="1219" alt="image" src="https://github.com/DataDog/chaos-controller/assets/17198797/847e733c-2f1c-45c1-aaa4-0f3ef5aa5ee6">

Reply:
I'm sorry I still don't get why we didn't needed this before and need this now 🤔

Reply:
prior, the filtering was done with an alternative rule, `except`. 
using `rules` is a different system that works a bit differently -- more open/flexible. Those rules are not compatible.
This [gitlab doc](https://docs.gitlab.com/ee/ci/jobs/job_control.html#types-of-manual-jobs) might help clear things up.